tools/importer-rest-api-specs/components/dataapigeneratorjson/models/api_types.go

```diff
@@ -0,0 +1,34 @@
+package models
+
+type ObjectDefinitionType string
+
+const (
+	BooleanObjectDefinitionType    ObjectDefinitionType = "Boolean"
+	CsvObjectDefinitionType        ObjectDefinitionType = "Csv"
+	DateTimeObjectDefinitionType   ObjectDefinitionType = "DateTime"
+	DictionaryObjectDefinitionType ObjectDefinitionType = "Dictionary"
+	IntegerObjectDefinitionType    ObjectDefinitionType = "Integer"
+	FloatObjectDefinitionType      ObjectDefinitionType = "Float"
+	ListObjectDefinitionType       ObjectDefinitionType = "List"
+	RawFileObjectDefinitionType    ObjectDefinitionType = "RawFile"
+	RawObjectObjectDefinitionType  ObjectDefinitionType = "RawObject"
+	ReferenceObjectDefinitionType  ObjectDefinitionType = "Reference"
+	StringObjectDefinitionType     ObjectDefinitionType = "String"
+
+	// Custom Types
+	EdgeZoneObjectDefinitionType                                ObjectDefinitionType = "EdgeZone"
+	LocationObjectDefinitionType                                ObjectDefinitionType = "Location"
+	SystemAssignedIdentityObjectDefinitionType                  ObjectDefinitionType = "SystemAssignedIdentity"
+	SystemAndUserAssignedIdentityListObjectDefinitionType       ObjectDefinitionType = "SystemAndUserAssignedIdentityList"
+	SystemAndUserAssignedIdentityMapObjectDefinitionType        ObjectDefinitionType = "SystemAndUserAssignedIdentityMap"
+	LegacySystemAndUserAssignedIdentityListObjectDefinitionType ObjectDefinitionType = "LegacySystemAndUserAssignedIdentityList"
+	LegacySystemAndUserAssignedIdentityMapObjectDefinitionType  ObjectDefinitionType = "LegacySystemAndUserAssignedIdentityMap"
+	SystemOrUserAssignedIdentityListObjectDefinitionType        ObjectDefinitionType = "SystemOrUserAssignedIdentityList"
+	SystemOrUserAssignedIdentityMapObjectDefinitionType         ObjectDefinitionType = "SystemOrUserAssignedIdentityMap"
+	UserAssignedIdentityListObjectDefinitionType                ObjectDefinitionType = "UserAssignedIdentityList"
+	UserAssignedIdentityMapObjectDefinitionType                 ObjectDefinitionType = "UserAssignedIdentityMap"
+	TagsObjectDefinitionType                                    ObjectDefinitionType = "Tags"
+	SystemDataObjectDefinitionType                              ObjectDefinitionType = "SystemData"
+	ZoneObjectDefinitionType                                    ObjectDefinitionType = "Zone"
+	ZonesObjectDefinitionType                                   ObjectDefinitionType = "Zones"
```

Comment:
we should document these

Reply:
These constants feel straightforward enough to not need to be documented. I'm not sure what we would document here other than:

```
        // TagsObjectDefinitionType specifies an ObjectDefinition of type Tags
	TagsObjectDefinitionType                                    ObjectDefinitionType = "Tags"
```

Is that what you are looking for or something more specific?

Reply:
it's more the complex use-cases (e.g. Csv, Dictionary, List and Custom Types) which are worth calling out:

```
// BooleanObjectDefinitionType specifies that this field contains a Boolean.
//
// ListObjectDefinitionType specifies that this field contains a List, where the List's Value Type will be defined as a Nested Object Definition.
//
// ZonesObjectDefinitionType specifies that this field contains Zones.
```

However if we're doing a subset we might as well do the lot for consistency, hence the 3 examples above - WDYT?